compiler/rustc_incremental/src/persist/file_format.rs

```diff
@@ -147,6 +163,14 @@ pub fn read_file(
         }
     }
 
+    // Check FILE_LEN
+    {
+        let mut len = [0u8; 8];
+        file.read_exact(&mut len)?;
+        let len_in_header = u64::from_le_bytes(len);
+        assert_eq!(len_in_header, fs::metadata(path)?.len());
```

Comment:
Should this be an ICE or a fatal error? Do we suggest a cargo clean?

Reply:
My personal suggestion (granted, as a non-contributor, so worth a grain of salt) would be to suggest a `cargo clean` with an ICE if possible. `cargo clean` does appear to unstick and allow later compilation to continue normally, and a fatal error doesn't seem appropriate/the right severity.

Reply:
If the assertion I'm trying to add in this PR ever fails, we have a new defect somewhere. We should ICE.

If we want to encourage people to paper over the problem with `cargo clean` we should just remove the file for them.